### ISSUE TITLE
Javadoc parsing fails (introduced in 3.15.14) causing StaticJavaParser.parse() to fail

### ISSUE THREAD

**Issue by edefazio** [javaparser/javaparser#2548]
While trying to read and create ASTs for an integration test
(reading all the Java source code from spring-core-5.1.9.RELEASE-sources.jar) the parser fails and throws this exception on trying to parse the file:
"spring-core-5.1.9.RELEASE-sources.jar\org\springframework\core\annotation\AnnotationUtils.java"

```java

com.github.javadocparser.TokenMgrError: Lexical error at line 6, column 6.  Encountered: "@" (64), after : "@"

	at com.github.javadocparser.GeneratedJavaDocParserTokenManager.getNextToken(GeneratedJavaDocParserTokenManager.java:708)
	at com.github.javadocparser.GeneratedJavaDocParser.jj_scan_token(GeneratedJavaDocParser.java:386)
	at com.github.javadocparser.GeneratedJavaDocParser.jj_3_8(GeneratedJavaDocParser.java:185)
	at com.github.javadocparser.GeneratedJavaDocParser.jj_3_7(GeneratedJavaDocParser.java:170)
	at com.github.javadocparser.GeneratedJavaDocParser.jj_2_7(GeneratedJavaDocParser.java:117)
	at com.github.javadocparser.GeneratedJavaDocParser.ContinueComment(GeneratedJavaDocParser.java:46)
	at com.github.javadocparser.GeneratedJavaDocParser.JavaDocLine(GeneratedJavaDocParser.java:33)
	at com.github.javadocparser.GeneratedJavaDocParser.Start(GeneratedJavaDocParser.java:14)
	at com.github.javadocparser.GeneratedJavaDocParser.parse(GeneratedJavaDocParser.java:8)
	at com.github.javaparser.GeneratedJavaParserTokenManager.CommonTokenAction(GeneratedJavaParserTokenManager.java:113)
	at com.github.javaparser.GeneratedJavaParserTokenManager.getNextToken(GeneratedJavaParserTokenManager.java:2968)
	at com.github.javaparser.GeneratedJavaParser.jj_ntk_f(GeneratedJavaParser.java:10577)
	at com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBody(GeneratedJavaParser.java:765)
	at com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceDeclaration(GeneratedJavaParser.java:410)
	at com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:156)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:123)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:155)
	at com.github.javaparser.JavaParser.parse(JavaParser.java:167)
	at com.github.javaparser.StaticJavaParser.parse(StaticJavaParser.java:99)
	at test.othertools.EclipseJDTTest.testFile(EclipseJDTTest.java:38)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
```

Note: this worked fine in earlier versions.  Also, I wanted to mention that perhaps we should consider having a better error messages, here, the error message says the exception occurrs on 
```java 
Lexical error at line 6, column 6.  Encountered: "@" (64), after : "@"
```
...but, this line number and column number is "relative" to the Javadoc String which is being parsed.... So in a large file with many members/Javadocs its hard to track down exactly where the error is... (I'm still doing research about which javadoc entry is causing the parse error) So, perhpas we should consider catching the parse error and wrapping/providing more detail (i.e. maybe the Positional range within the file where the exception occurs) so that these types of errors are easier to identify

**Comment by edefazio** [comment]
Here's how to reproduce (I extracted the individual file from spring.jar just to isolate it and ran this test)
```java
    public void testFile() throws FileNotFoundException {
        StaticJavaParser.parse( new FileInputStream("C:\\temp\\AnnotationUtils.java") );
    } 
```
(now looking to see if I can figure out which javadoc within the file is the offender)

**Comment by edefazio** [comment]
Looks like this particular javadoc is causing the issue:
```java
	/**
	 * Determine whether the specified method has searchable annotations,
	 * i.e. not just {@code java.lang} or {@code org.springframework.lang}
	 * annotations such as {@link Deprecated} and {@link Nullable}.
	 * @param ifcMethod the interface method to check
	 * @@since 5.0.5
	 */
	private static boolean hasSearchableAnnotations(Method ifcMethod) {
```

specifically the `@@since` seems to cause this problem

**Comment by matozoid** [comment]
Okay, this wasn't supposed to be in the main parsing path yet.

**Comment by matozoid** [comment]
Ah, in java.jj the Javadoc parser is called. Hadn't seen that (the PR was kind of big) so I'll disable that right now.

**Comment by edefazio** [comment]
was thinking of closing, but I think the underlying Javadoc parsing issue should probably looked into
...i.e. these (2) cases seem to cause issues:

- a javadoc with `/**` within it causes a parse error  (i.e. `/**'{@code /docs/**}' `

- a javadoc with `@@` (i.e. `/**@@since*/`

**Comment by matozoid** [comment]
The normal parsing path does not parse Javadoc. It certainly shouldn't suddenly start doing so with a parser that isn't finished.

**Comment by matozoid** [comment]
Alright, the whole PR was pretty much half-baked. I'm going to revert it and put it on a [DRAFT] branch.

**Comment by matozoid** [comment]
Fixed in master, but still a problem in #2550
